include/mrdocs/Metadata/Alias.hpp

```diff
@@ -0,0 +1,46 @@
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// Copyright (c) 2024 Fernando Pelliccioni (fpelliccioni@gmail.com)
+//
+// Official repository: https://github.com/cppalliance/mrdocs
+//
+
+#ifndef MRDOCS_API_METADATA_NAMESPACEALIAS_HPP
+#define MRDOCS_API_METADATA_NAMESPACEALIAS_HPP
+
+#include <mrdocs/Platform.hpp>
+#include <mrdocs/Metadata/Info.hpp>
+#include <mrdocs/Metadata/Source.hpp>
+#include <mrdocs/Metadata/Type.hpp>
+#include <utility>
+
+namespace clang {
+namespace mrdocs {
+
+/** Info for namespace aliases.
+*/
+struct AliasInfo
+    : IsInfo<InfoKind::Alias>
+    , SourceInfo
+{
+    /** Aliased symbol. */
+    SymbolID AliasedSymbol = SymbolID::invalid;
+
+    /** The fully qualified name of the alias. */
+    std::unique_ptr<NameInfo> FullyQualifiedName;
```

Comment:
You only need to store the `NameInfo` -- it will contain the `SymbolID` of the referenced symbol (so you should remove `AliasedSymbol`). Also `FullyQualifiedName` should be renamed `AliasedSymbol`.

Reply:
How this piece of code have to look after the changes in the Alias struct?

```
class NamespaceAliasBlock
    : public TopLevelBlock<AliasInfo>
{
public:
    using TopLevelBlock::TopLevelBlock;
    Error
    parseRecord(
        Record const& R,
        unsigned ID,
        llvm::StringRef Blob) override
    {
        switch(ID)
        {
        case NAMESPACE_ALIAS_SYMBOL:
            return decodeRecord(R, I->AliasedSymbol, Blob);
        default:
            return TopLevelBlock::parseRecord(R, ID, Blob);
        }
        return TopLevelBlock::parseRecord(R, ID, Blob);
    }
    ```

Reply:
`NamespaceAliasBlock` was renamed to `AliasBlock`